apisix/plugins/chaitin-waf.lua

```diff
@@ -224,33 +242,37 @@ end
 
 
 local function check_match(conf, ctx)
-    local match_passed = true
+    if not conf.match or #conf.match == 0 then
+        return true
+    end
 
-    if conf.match then
-        for _, match in ipairs(conf.match) do
-            -- todo: use lrucache to cache the result
-            local exp, err = expr.new(match.vars)
-            if err then
-                local msg = "failed to create match expression for " ..
+    for _, match in ipairs(conf.match) do
+        local cache_key = core.json.encode(match.vars)
```

Comment:
I think using the index or address(match.vars) of an array as the cache_key would be better.
* ctx.conf_id .. # .. i
* tostring(match.vars)